systemtest/src/test/java/io/strimzi/systemtest/specific/RackAwarenessST.java

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.systemtest.specific;
+
+import io.fabric8.kubernetes.api.model.Affinity;
+import io.fabric8.kubernetes.api.model.NodeAffinity;
+import io.fabric8.kubernetes.api.model.NodeSelectorRequirement;
+import io.fabric8.kubernetes.api.model.Pod;
+import io.fabric8.kubernetes.api.model.PodAffinityTerm;
+import io.strimzi.api.kafka.model.KafkaConnectResources;
+import io.strimzi.api.kafka.model.KafkaMirrorMaker2Resources;
+import io.strimzi.api.kafka.model.KafkaResources;
+import io.strimzi.api.kafka.model.Rack;
+import io.strimzi.systemtest.AbstractST;
+import static io.strimzi.systemtest.Constants.CONNECT;
+import static io.strimzi.systemtest.Constants.MIRROR_MAKER2;
+import static io.strimzi.systemtest.Constants.REGRESSION;
+import io.strimzi.systemtest.annotations.ParallelNamespaceTest;
+import io.strimzi.systemtest.annotations.ParallelSuite;
+import static io.strimzi.systemtest.resources.ResourceManager.kubeClient;
+import io.strimzi.systemtest.resources.crd.KafkaConnectResource;
+import io.strimzi.systemtest.storage.TestStorage;
+import io.strimzi.systemtest.templates.crd.KafkaConnectTemplates;
+import io.strimzi.systemtest.templates.crd.KafkaMirrorMaker2Templates;
+import io.strimzi.systemtest.templates.crd.KafkaTemplates;
+import io.strimzi.systemtest.utils.StUtils;
+import io.strimzi.systemtest.utils.kafkaUtils.KafkaConnectUtils;
+import io.strimzi.systemtest.utils.kubeUtils.objects.PodUtils;
+import static io.strimzi.test.k8s.KubeClusterResource.cmdKubeClient;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasEntry;
+import static org.hamcrest.Matchers.is;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+@Tag(REGRESSION)
+@ParallelSuite
+class RackAwarenessST extends AbstractST {
+    private static final Logger LOGGER = LogManager.getLogger(RackAwarenessST.class);
+    private static final String TOPOLOGY_KEY = "kubernetes.io/hostname";
+
+    @ParallelNamespaceTest
+    void testKafkaRackAwareness(ExtensionContext extensionContext) {
+        TestStorage storage = storageMap.get(extensionContext);
+        String namespace = storage.getNamespaceName();
+        String clusterName = storage.getClusterName();
```

Comment:
So, what about this? In all other tests we are using just the `TestStorage`. The idea why was `TestStorage` created is, that we wanted to get rid of all kind of variables inside the tests (variables like `clusterName`, `namespaceName`, `kafkaLabelSelector` etc.). When you create new variables, to which you pass the `TestStorage` stuff, you will have two duplicates, which is not correct FMPOV.

So, how I mentioned earlier, you have two options:

1. use the `TestStorage` and the getters
2. remove the `TestStorage` usage and pass it directly to the variables, to not have duplicates etc.